metals/src/main/scala/scala/meta/internal/metals/CompilerPlugins.scala

```diff
@@ -37,42 +37,45 @@ import scala.meta.io.AbsolutePath
  * - enabling the plugin does not break the presentation compiler in unexpected ways.
  */
 class CompilerPlugins {
-  private val cache = TrieMap.empty[AbsolutePath, Boolean]
+  private val cache = TrieMap.empty[Seq[AbsolutePath], Boolean]
 
   def filterSupportedOptions(options: Seq[String]): Seq[String] = {
-    options.filter { option =>
-      if (option.startsWith("-Xplugin:")) {
-        val path = AbsolutePath(option.stripPrefix("-Xplugin:"))
-        cache.getOrElseUpdate(path, isSupportedPlugin(path))
-      } else if (option.startsWith("-P:")) {
-        isSupportedPlugin.exists(plugin => option.startsWith(s"-P:$plugin:"))
-      } else {
-        true
-      }
+    options.filter {
+      case s"-Xplugin:$pathsString" =>
+        val paths = pathsString.split(",").map(AbsolutePath(_)).toSeq
+        cache.getOrElseUpdate(paths, isSupportedPlugin(paths))
+      case s"-P:$plugin:$_" =>
+        supportedPlugins.contains(plugin)
+      case _ => true
     }
   }
 
-  private val isSupportedPlugin = Set(
+  private val supportedPlugins = Set(
     "kind-projector", // https://github.com/non/kind-projector
     "bm4", // https://github.com/oleg-py/better-monadic-for
     // Intentionally not supported:
     // "macro-paradise-plugin", see https://github.com/scalameta/metals/issues/622
   )
 
-  private def isSupportedPlugin(path: AbsolutePath): Boolean = {
-    path.isJar && {
-      try {
-        FileIO.withJarFileSystem(path, create = false, close = true) { root =>
-          val xml = XML.load(
-            Files.newInputStream(root.resolve("scalac-plugin.xml").toNIO)
-          )
-          val name = (xml \ "name").text
-          isSupportedPlugin(name)
+  private def isSupportedPlugin(paths: Seq[AbsolutePath]): Boolean = {
+    paths.exists { path =>
```

Comment:
It seems like Metals will accept unsupported plugin in the following case like `-Xplugin:bm4,invalid-plugin` because `bm4` is supported plugin, and `paths.exists` will be true in this case.

Reply:
In Mill we use `-Xplugin:` with comma-separated list of paths for the classpath to run that specific plugin. I don't know if you can pass multiple plugins on the same `-Xplugin:` list. If you can, how can handle this scenario?

Reply:
> I don't know if you can pass multiple plugins on the same -Xplugin: list.

So I don't think there is anything keeping someone from doing that. I was curious in Scala 3 so I took a look [here](https://github.com/lampepfl/dotty/blob/b461c943dbe47caf19f975833d686d8e666b5285/compiler/src/dotty/tools/dotc/plugins/Plugins.scala#L28-L45) and if I'm understanding what is happening correctly that classpath _could_ in theory hold multiple different plugins. Most often that's not the case and instead it's just the classpath necessary for a single plugin to work (I do the same thing in [sbt-scoverage](https://github.com/scoverage/sbt-scoverage/blob/e4a5bc7d5c3bc66c9f18a1b7ef73a54081b64858/src/main/scala/scoverage/ScoverageSbtPlugin.scala#L151)), but in theory I think this could be two independent plugins.

The tricky part is how would you know what part of the classpath is supposed to be for the plugin vs another plugin. I have a feeling that this is incredibly niche.

> It seems like Metals will accept unsupported plugin in the following case like -Xplugin:bm4,invalid-plugin because bm4 is supported plugin, and paths.exists will be true in this case.

@tanishiking out of curiosity, have you seen plugins defined like this out in the wild?

Reply:
The plugin I had in mind was actually `sbt-scoverage`, but, right, sbt-scoverage case is "part of the classpath is supposed to be for the plugin" and they should be valid. (I overlooked this case even though `sbt-scoverage` was in my mind 😅 )

I was a bit worried at first, what if users somehow sneak in the invalid plugin this way, but now I have a second thought that it's indeed too niche.

So, never mind, about my first comment!